mcx/jax.py

```diff
@@ -87,12 +94,19 @@ def _update_progress_bar(iter_num):
         number is a multiple of the print_rate
         """
         _ = lax.cond(
-            iter_num % print_rate == 0,
+            (iter_num % print_rate == 0) & (iter_num != num_samples - remainder),
```

Comment:
This (and the few lines below) update tqdm by `print_rate` amount except for the last time. The last time you need to add the remainder. This fixes the bug that happens if `num_samples` isn't a multiple of 20 (ex: run the sampler for 1010 iterations)